xwakes/wit/component.py

```diff
@@ -385,3 +510,895 @@ def get_shorthand_type(self) -> str:
+        aux = np.sqrt((1 - f_roi_level) / f_roi_level)
+
+        return (aux + np.sqrt(aux**2 + 4*q**2))*f_r/(2*q) - f_r
+
+
+class ComponentClassicThickWall(Component):
+    def __init__(self,
+                 kind: str = None,
+                 plane: str = None,
+                 exponents: Tuple[int, int, int, int] | None = None,
+                 source_exponents: Tuple[int, int] | None = None,
+                 test_exponents: Tuple[int, int] | None = None,
+                 layer: Layer = None,
+                 radius: float = None,
+                 resistivity: float = None,
+                 factor: float = 1.0,
+                 length: float = 1.0,
+                 zero_rel_tol: float = 0.01
+                 ) -> ComponentClassicThickWall:
+        """
+        Creates a single component object modeling a resistive wall
+        impedance/wake, based on the "classic thick wall formula" (see e.g.
+        A. W. Chao, chap. 2 in "Physics of Collective Beams Instabilities in
+        High Energy Accelerators", John Wiley and Sons, 1993).
+        Only longitudinal and transverse dipolar impedances are supported here.
+        :param plane: the plane the component corresponds to
+        :param exponents: four integers corresponding to (source_x, source_y,
+        test_x, test_y) aka (a, b, c, d)
+        :param layer: the chamber material, as a wit Layer object
+        :param radius: the chamber radius in m
+        :return: A component object
+        """
+
+        if (layer is None and resistivity is None):
+            raise ValueError(
+                "Either a layer object or a resistivity must be provided")
+        if layer is not None and resistivity is not None:
+            raise ValueError(
+                "Both a layer object and a resistivity cannot be provided")
+
+        if radius is None:
+            raise ValueError("Radius must be provided")
+
+        self.layer = layer
+        self.radius = radius
+        self.plane = plane
+        self.factor = factor
+        self.length = length
+        self.resistivity = resistivity
+        self.zero_rel_tol = zero_rel_tol
+
+        source_exponents, test_exponents, plane = _handle_plane_and_exponents_input(
+                                    kind=kind, exponents=exponents,
+                                    source_exponents=source_exponents,
+                                    test_exponents=test_exponents,
+                                    plane=plane)
+
+        # we set impedance and wake to a dummy callable because they will be
+        # overridden by methods
+        super().__init__(impedance=lambda x: 0, wake=lambda x: 0, plane=plane,
+                         source_exponents=source_exponents,
+                         test_exponents=test_exponents,
+                         name="Classic Thick Wall")
+
+    def impedance(self, f):
+        layer = self.layer
+        material_magnetic_permeability = mu0
```

Comment:
This is less general than it could be (and that it used to be...): one could include constant permeability (or, equivalently, magnetic susceptibility) here.